app/src/components/LoginForm.js

```diff
@@ -1,55 +1,106 @@
 import React from 'react';
-import TextField from '@material-ui/core/TextField';
-import FormControl from '@material-ui/core/FormControl';
+import { withStyles } from '@material-ui/core/styles';
 import Button from '@material-ui/core/Button';
 import { compose, withHandlers, withState } from 'recompose';
+import FormControl from './FormControl';
+import Notice from './Notice';
+import errorForField from '../utils/errorForField';
+import validateLogin from '../validators/validateLogin';
 
-const handleLogin = ({ email, loginMutation, password }) => () => (
-  loginMutation({ email, password })
-);
+const styles = theme => ({
+  button: {
+    float: 'right',
+    marginTop: theme.spacing.unit
+  },
+  formLoading: {
+    visibility: 'hidden'
+  },
+  form: {
+    maxWidth: 400,
+    width: '80%',
+    marginTop: theme.spacing.unit * 8,
+    marginBottom: theme.spacing.unit * 2,
+  },
+  formControl: {
+    marginBottom: theme.spacing.unit
+  }
+});
+
+const handleLogin = ({ email, loginMutation, setErrors, password }) => async event => {
+  event.preventDefault();
+  const errors = validateLogin({ email, password });
+  setErrors(errors);
+  if (errors) { return; }
+
+  const result = await loginMutation({ email, password });
+  if (result.errors && result.errors.length) {
+    return setErrors(result.errors.map(error => ({ message: error })));
+  }
+};
 
-const onChangeEmail = ({ setEmail }) => ({ target: { value }}) => {
+const onChangeEmail = ({ errors, setEmail, setErrors }) => ({ target: { value }}) => {
+  if (errorForField(errors, 'email') || errorForField(errors)) {
+    setErrors(null);
```

Comment:
on change clear input error or any server error that would be disabling the submit button